### PR TITLE
feat&chore: new enemy (venomite) and drone prefab updated

### PR DESCRIPTION
**Pull request by markmeliaa**
This brings a new enemy to the table, the venomites! Their behaviour is the following:´
- They will be patrolling a certain area.
- When they see you from afar, they will start shooting you.
- They will keep shooting you until you run away from them or you reduce their life to the 50%.
- Once their life is reduced to 50% or less, they will go towards you and start attacking with their baton.
- They will follow you until they kill you or you kill them.

Aside from that, the drone was updated with all the new improvements that were developed during this VS, if you want more information, check #534 .

### CODE REVIEW

**Review by DADSILVER — COMMENTED** [review]
Only 1 question

**Review by DADSILVER — APPROVED** [review]

File: Game/Scripts/SeekBehaviourScript.h [hunk 1]
```diff
@@ -27,11 +27,5 @@ class SeekBehaviourScript : public Script
...
 	ComponentTransform* targetTransform;
 	ComponentRigidBody* ownerRigidBody;
-	GameObject* ownerRigidBodyGO;
 	ComponentTransform* ownerTransform;
-};
-
-inline GameObject* SeekBehaviourScript::GetTarget() const
-{
-	return target;
-}
```
[the remark this answers is not in the document]
Comment (markmeliaa) [reply Game/Scripts/SeekBehaviourScript.h:37]: Well, currently (afaik) we are not using inline functions in the gameplay scripts, that why I deleted that one, to have coherence with the rest of the code. We could discuss about making inline functions in the gameplay scripts, I'm not against it.

File: Game/Scripts/EnemyVenomiteScript.cpp [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+	GameObject* seekTarget = seekScript->GetTarget();
+
+	if (seekTarget)
+	{
+		ComponentTransform* seekTargetTransform = seekTarget->GetComponent<ComponentTransform>();
+
+		if (venomiteState != VenomiteBehaviours::PATROL)
+		{
+			venomiteState = VenomiteBehaviours::PATROL;
+			patrolScript->StartPatrol();
```
Comment (PaablooCH) [inline Game/Scripts/EnemyVenomiteScript.cpp:73]: move this inside of this `if`:
```
if (patrolScript && venomiteState == VenomiteBehaviours::PATROL)
{
	batonGameObject->Disable();
	blasterGameObject->Disable();

	patrolScript->Patrolling();

	componentAnimation->SetParameter("IsRunning", true);
}
```

**Review by PaablooCH — CHANGES_REQUESTED** [review]
🇬🇧 : nice, i hope we can deliver all that great code soon, just 1 thing:
🇪🇸 : suélteme el brazo señor:

**Review by PaablooCH — APPROVED** [review]
🇬🇧 👍 
🇪🇸 no más

**Comment by markmeliaa** [comment]
> 🇬🇧 👍 
> 🇪🇸 no más

🇬🇧 👍 
🇪🇸 me caes regular

**Pull request merged**